spirv_msl.cpp

```diff
@@ -398,6 +515,20 @@ void CompilerMSL::emit_header()
 	statement("");
 	statement("using namespace metal;");
 	statement("");
+	emit_msl_defines();
+}
+
+void CompilerMSL::emit_msl_defines()
+{
+	statement("// Standard GLSL->MSL redefinitions");
+	statement("#define discard discard_fragment()");
```

Comment:
Discard would be a terminator, so it indeed does not hit emit_instruction.
However, these kinds of overrides would neatly fit into the BackendVariations, as it's intended to resolve trivial differences between target languages.

Adding a `std::string discard = "discard";` there could solve this. MSL can then override that to `discard_fragment()`.

For the other cases, these are all handled by emit_instruction.